incubator/hnc/pkg/reconcilers/hierarchical_namespace.go

```diff
@@ -30,20 +33,74 @@ type HierarchicalNamespaceReconciler struct {
 }
 
 // Reconcile sets up some basic variables and then calls the business logic.
-// It currently only generates some logs for testing.
+// It currently handles basic creation of namespace and hierarchyconfig.
 func (r *HierarchicalNamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// TODO report error state if the webhook is bypassed - see issue
 	//  https://github.com/kubernetes-sigs/multi-tenancy/issues/459
 	if ex[req.Namespace] {
 		return ctrl.Result{}, nil
 	}
 
+	ctx := context.Background()
 	log := r.Log.WithValues("trigger", req.NamespacedName)
 	log.Info("Reconciling HNS")
 
+	// Names of the self-serve subnamespace and the parent namespace.
+	nm := req.Name
+	pnm := req.Namespace
+
+	// We want to trigger the hierarchyConfig reconciliation on the self-serve subnamespace, since
+	// the parent's 'requiredChildren' field will be deprecated. The HierarchyConfig Reconciler
+	// will do the real job of updating the hierarchy in the forest and creating namespaces and
+	// hierarchyconfig (hc) instances.
+	//
+	// Please note that the self-serve subnamespace and its hc object don't exist on apiserver for now.
+	// Just enqueuing an in-memory hc instance for hc reconciliation will lose the hc's parent field
+	// since the hc reconciler will get nothing from the apiserver and create an empty in-memory hc.
```

Comment:
Hmm, this isn't what I had in mind. I thought we were going to do this:

* In the HNS reconciler:
  * Create the child namespace in the forest (_not_ on the apiserver) and set its owner to the parent namespace
  * Enqueue the child namespace for reconciliation
* In the HC reconciler:
  * Notice that the child namespace doesn't exist on the apiserver, but that it _does_ exist in the forest
  * Create all apiserver objects

This centralizes all the writing of HCs and namespaces in one place (the HC reconciler) so the HNS reconciler only needs to read/write the HNS.

Does this make sense? Or is there a reason this doesn't work?